tools/bpm_bench/bpm_bench.cpp

```diff
@@ -100,6 +96,45 @@ struct BpmMetrics {
   }
 };
 
+struct BustubBenchPageHeader {
+  uint64_t seed_;
+  uint64_t page_id_;
+  char data_[0];
+};
+
+/// Modify the page and save some data inside
+auto ModifyPage(char *data, size_t page_idx, uint64_t seed) -> void {
+  auto pg = reinterpret_cast<BustubBenchPageHeader *>(data);
+  pg->seed_ = seed;
+  pg->page_id_ = page_idx;
+  pg->data_[pg->seed_ % 4000] = pg->seed_ % 256;
+}
+
+/// Check the page and verify the data inside
+auto CheckPageConsistentNoSeed(const char *data, size_t page_idx) -> void {
+  auto pg = reinterpret_cast<const BustubBenchPageHeader *>(data);
```

Comment:
Seems `auto` will drop the `const` qualifier?
```suggestion
  const auto *pg = reinterpret_cast<const BustubBenchPageHeader *>(data);
```